core/validation/specific_validations/towns_fund_round_four.py

```diff
@@ -16,22 +18,43 @@ def validate(workbook: dict[str, pd.DataFrame]) -> list["TownsFundRoundFourValid
     validations = (validate_project_risks, validate_programme_risks)
 
     validation_failures = []
-    for validation_funct in validations:
-        validation_failures.extend(validation_funct(workbook))
+    for validation_func in validations:
+        failures = validation_func(workbook)
+        if failures:
+            validation_failures.extend(failures)
 
     return validation_failures
 
 
-def validate_project_risks(workbook: dict[str, pd.DataFrame]) -> list["TownsFundRoundFourValidationFailure"]:
-    # TODO: validate there is a risk for each project.
-    return []
+def validate_project_risks(workbook: dict[str, pd.DataFrame]) -> list["TownsFundRoundFourValidationFailure"] | None:
+    """Validates that each project has at least one Risk row associated with it.
+
+    :param workbook: A dictionary where keys are sheet names and values are pandas
+                     DataFrames representing each sheet in the Round 4 submission.
+    :return: ValidationErrors
+    """
+    all_project_ids = workbook["Project Details"]["Project ID"]
+    risk_project_ids = workbook["RiskRegister"]["Project ID"]
+    projects_missing_risks = set(all_project_ids).difference(risk_project_ids)
+
+    if projects_missing_risks:
+        project_numbers = [int(project_id.split("-")[2]) for project_id in projects_missing_risks]
+        return [
+            TownsFundRoundFourValidationFailure(
+                tab="Risk Register",
+                section=f"Project Risks - Project {project}",
```

Comment:
I think this might be a bit of an issue because for NonUniqueCompositeKeyFailure we have:
```
section = f"Project {project_number} Risks"
```
so the user could end up with a sub-section for "Project 1 Risks" and "Projects Risks - Project 1"

Reply:
Nice spot I'll modify to existing wording

Reply:
I think I prefer this way of writing it though...

Reply:
haha ok I'll swap it to the other way

Reply:
Done 👍